keyboards/dc01/left/matrix.c

```diff
@@ -387,37 +387,13 @@ static void unselect_cols(void)
 
 // Complete rows from other modules over i2c
 i2c_status_t i2c_transaction(uint8_t address, uint32_t mask, uint8_t col_offset) {
-    i2c_status_t status = i2c_start(address, 5);
-    if (status < 0) {
-        goto error;
-    }
-
-    status = i2c_write(0x01, 50);
-    if (status < 0) {
-        goto error;
-    }
-
-    status = i2c_start(address | I2C_READ, 50);
+    uint8_t data[MATRIX_ROWS];
+    i2c_status_t status = i2c_receive(address, data, MATRIX_ROWS, 5+(50*(MATRIX_ROWS+3)));
```

Comment:
The original sequence of operations matches `i2c_readReg()`.

And the calculated timeout is probably too large — the original implementation uses a 5 ms timeout on `i2c_start()`, but 50 ms timeouts on the actual byte transfers; this cannot be replicated in the high-level I2C API.  The issue here is that a large timeout will make the keyboard unusable if some module is not connected (the custom code for this keyboard does not have any logic for throttling requests to a module after getting an excessive number of errors).

This is actually another major difference between the AVR and ChibiOS implementations of the I2C master API — the AVR implementation will retry in `i2c_start()` after failing to get ACK for the slave address, the ChibiOS implementation will just try to perform the transaction once.  Also in the AVR implementation the `timeout` argument gets applied individually to every transferred byte, while ChibiOS applies the timeout to the transaction as a whole.
```suggestion
    i2c_status_t status = i2c_readReg(address, 0x01, data, MATRIX_ROWS, 5);
```